modules/core/src/main/java/org/apache/ignite/configuration/CacheConfiguration.java

```diff
@@ -1194,14 +1194,15 @@ public CacheRebalanceMode getRebalanceMode() {
      * Gets cache rebalance order. Rebalance order can be set to non-zero value for caches with
      * {@link CacheRebalanceMode#SYNC SYNC} or {@link CacheRebalanceMode#ASYNC ASYNC} rebalance modes only.
```

Comment:
I think we should also mention for SYNC, ASYNC cache mode in javadoc something like 'SYNC caches with the same order will always be rebalanced first'

Reply:
Well, IMHO it is already mentioned below:

>      * Note that a cache with {@link CacheRebalanceMode#SYNC SYNC} rebalancing mode always takes precedence
>      * over caches with {@link CacheRebalanceMode#ASYNC ASYNC} rebalancing mode
>      * in the group of caches with the same rebalance order. This means caches with {@link CacheRebalanceMode#SYNC SYNC}
>      * rebalancing mode will be rebalanced in the first place.

Do you prefer that this fact should be highlighted in the very beginning?

Reply:
I think so since the `order` has complex meaning now.

Reply:
I updated the Javadoc as follows:

>      * Gets cache rebalance order. Rebalance order can be set to non-zero value for caches with
>      * {@link CacheRebalanceMode#SYNC SYNC} or {@link CacheRebalanceMode#ASYNC ASYNC} rebalance modes only.
>      * Note that caches with {@link CacheRebalanceMode#SYNC SYNC} rebalancing mode are always rebalanced prior to caches
>      * with {@link CacheRebalanceMode#ASYNC ASYNC} rebalancing mode when rebalancing order is the same.
>      * <p/>
>      * The rebalance order guarantees that rebalancing for this cache will start only when rebalancing for
>      * all caches with smaller rebalance order will be completed.
>      * <p/>
>      * If not set, cache order is 0.

Does it look good enough?